service/src/main/java/co/cask/wrangler/dataset/workspace/WorkspaceDataset.java

```diff
@@ -128,6 +128,28 @@ public void createWorkspaceMeta(String id, String name, DataType type,
   }
 
   /**
+   * Checks if a workspace exists.
+   *
+   * @param id of the workspace to be checked for.
+   * @return true if workspace exists, false otherwise.
+   * @throws WorkspaceException thrown if there are any issues with workspace.
+   */
+  public boolean hasWorkspace(String id) throws WorkspaceException {
+    try {
+      Row row = table.get(toKey(id));
+      if (row != null) {
```

Comment:
Addressed comment -- not sure why it worked. But using isEmpty is right way to do it.